chalicelib/checks/deployment_checks.py

```diff
@@ -0,0 +1,189 @@
+        if success:
+            action.status = 'DONE'
+            action.output = 'Successfully triggered termination of indexer env %s' % env_to_terminate
+        else:
+            action.status = 'FAIL'
+            action.output = 'Encountered an error while issuing terminate call, please check the AWS EB Console'
+    else:
+        action.status = 'FAIL'
+        action.output = 'Did not get an env_to_terminate from the associated check - bad action call?'
+
+    return action
+
+
+@check_function(env='fourfront-hotseat', application_version=None)
+def provision_indexer_environment(connection, **kwargs):
+    """ Provisions an indexer environment for the given env. Note that only one indexer can be online
+        per application (one for 4DN, one for CGAP).
+
+        IMPORTANT: env is an EB ENV NAME ('data', 'staging' are NOT valid).
+        IMPORTANT: This ecosystem will break some integrated tests in dcicutils
+    """
+    check = CheckResult(connection, 'provision_indexer_environment')
+    env = kwargs.get('env', None)
+    application_version = kwargs.get('application_version', None)
+
+    if application_version is None:
+        check.status = 'ERROR'
+        check.summary = 'Did not provide application_version to deploy indexer, which is required. ' \
+                        'Get this information from the EB Console.'
+        return check
+
+    def _deploy_indexer(e, version):
+        if is_fourfront_env(e):
+            description = try_to_describe_indexer_env(FF_ENV_INDEXER)
+        else:
+            description = try_to_describe_indexer_env(CGAP_ENV_INDEXER)
+        if description is not None:
+            check.status = 'ERROR'
+            check.summary = 'Tried to spin up indexer env for %s when one already exists for this portal' % e
+            return False
+        else:
+            return EBDeployer.deploy_indexer(e, version)
+
+    if is_cgap_env(env) or is_fourfront_env(env):
+        success = _deploy_indexer(env, application_version)
+        if success:
+            check.status = 'PASS'
+            check.summary = 'Successfully triggered indexer-server provision for environment %s' % env
+        else:
+            check.status = 'ERROR'
+            check.summary = 'An error occurred on deployment. Check the AWS EB Console.'
+    else:
+        check.status = 'FAIL'
+        check.summary = 'Gave an unknown environment: %s' % env
+
+    return check
+
+
+@check_function(env='fourfront-mastertest',
+                branch='master',
+                application_version_name=None, repo=None)
+def deploy_application_to_beanstalk(connection, **kwargs):
+    """ Deploys application to beanstalk under the given application_version name + a branch
+
+        NOTE: CGAP requires kwargs!
+    """
+    check = CheckResult(connection, 'deploy_application_to_beanstalk')
+    env = kwargs.get('env', 'fourfront-mastertest')  # by default
```

Comment:
What facility would set these to other than the default values?

Reply:
`app.run_check_or_action(connection, <check_name>, {'env': 'fourfront-green'})`.
I think once a check has been run it's arguments in the decorator will show up on the UI.

Reply:
Correct on arguments showing up on UI - I'm not positive but I think including the arguments in the decorator obviates the need for having a default in the .get request eg. here: kwargs.get('branch', 'master')
Is it still true that you need to run a check locally with primary=True to get it to show up in the UI?  If so then believe I'm correct but worth a test.